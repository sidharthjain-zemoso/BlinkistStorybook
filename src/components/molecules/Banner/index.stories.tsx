import { ComponentStory } from "@storybook/react";
import CardBanner from "./index";

export default {
  title: "Molecules/Card Banner",
  component: CardBanner,
};

const Template: ComponentStory<typeof CardBanner> = (args) => (
  <CardBanner {...args} />
);

export const cardBanner = Template.bind({});
cardBanner.args = {};
