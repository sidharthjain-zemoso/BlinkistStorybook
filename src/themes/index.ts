import { createTheme } from "@mui/material/styles";

declare module "@mui/material/Typography" {
  interface TypographyPropsVariantOverrides {
    heading: true;
    heading3: true;
    subtitle3: true;
    body3: true;
    caption1: true;
    caption2: true;
    caption3: true;
  }
}

declare module "@mui/material/styles" {
  interface TypographyVariants {
    heading: React.CSSProperties;
    heading3: React.CSSProperties;
    subtitle3: React.CSSProperties;
    body3: React.CSSProperties;
    caption1: React.CSSProperties;
    caption2: React.CSSProperties;
    caption3: React.CSSProperties;
  }

  interface TypographyVariantsOptions {
    heading?: React.CSSProperties;
    heading3: React.CSSProperties;
    subtitle3?: React.CSSProperties;
    body3?: React.CSSProperties;
    caption1?: React.CSSProperties;
    caption2?: React.CSSProperties;
    caption3: React.CSSProperties;
  }

  interface PaletteColor {
    darker?: string;
    border?: string;
    placeholder?: string;
  }

  interface SimplePaletteColorOptions {
    darker?: string;
    placeholder?: string;
    border?: string;
  }

  interface Palette {
    greyscale: Palette["primary"];
    textcolor: Palette["primary"];
    backgroundcolor: Palette["primary"];
  }

  interface PaletteOptions {
    greyscale?: PaletteOptions["primary"];
    textcolor?: PaletteOptions["primary"];
    backgroundcolor?: PaletteOptions["primary"];
  }
}

export const theme = createTheme({
  components: {
    MuiButtonBase: {
      styleOverrides: {
        root: {
          // "&:hover": {
          //   backgroundColor: "none",
          // },
        },
      },
    },
  },
  spacing: 4,
  palette: {
    primary: {
      light: "#E2F2E9",
      main: "#2CE080",
      dark: "#22C870",
      darker: "#20BA68",
    },
    secondary: {
      light: "#DFE8F6",
      main: "#0365F2",
    },
    greyscale: {
      light: "#BAC9CF",
      main: "#6C787F",
      dark: "#3A4649",
      darker: "#042330",
    },
    textcolor: {
      light: "#6D787E",
      main: "#03314B",
      placeholder: "#747575",
      dark: "#042330",
    },
    backgroundcolor: {
      light: "#F7F7F7",
      main: "#F1F6F4",
    },
  },
  typography: {
    heading: {
      fontSize: "36px",
      lineHeight: "45px",
      // fontWeight: 700,
      fontFamily: "Cera Pro Medium",
    },
    heading3: {
      fontSize: "24px",
      lineHeight: "30.17px",
      // fontWeight: 700,
      fontFamily: "Cera Pro Medium",
    },
    subtitle1: {
      fontSize: "24px",
      lineHeight: "32px",
      // fontWeight: 500,
      fontFamily: "Cera Pro Medium",
    },
    subtitle2: {
      fontSize: "18px",
      lineHeight: "24px",
      fontWeight: 700,
      fontFamily: "Cera Pro Light",
    },
    subtitle3: {
      fontSize: "18px",
      lineHeight: "24px",
      // fontWeight: 400,
      fontFamily: "Cera Pro Light",
    },
    body1: {
      fontSize: "16px",
      lineHeight: "20px",
      // fontWeight: 700,
      fontFamily: "Cera Pro Medium",
    },
    body2: {
      fontSize: "16px",
      lineHeight: "24px",
      // fontWeight: 400,
      fontFamily: "Cera Pro Light",
    },
    body3: {
      fontSize: "22px",
      lineHeight: "20px",
      // fontWeight: 400,
      fontFamily: "Cera Pro Light",
    },
    caption1: {
      fontSize: "14px",
      lineHeight: "22px",
      // fontWeight: 400,
      fontFamily: "Cera Pro Light",
    },
    caption2: {
      fontSize: "14px",
      lineHeight: "22px",
      // fontWeight: 700,
      fontFamily: "Cera Pro Medium",
    },
    caption3: {
      fontSize: "14px",
      lineHeight: "17.6px",
      // fontWeight: 400,
      fontFamily: "Cera Pro Light",
    },
  },
});

// theme.overrides = {
//   ...theme.overrides,
//   MuiButton: {
//     root: {
//     }
//   }
// }
