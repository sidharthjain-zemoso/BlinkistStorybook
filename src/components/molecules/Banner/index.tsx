import { Grid } from "@mui/material";
import React from "react";
import Logo from "../../atoms/Logo";
import TypographyText from "../../atoms/Typography";
import { theme } from "../../../themes";

type CardBannerProps = {
  sx?: Object;
};

const CardBanner: React.FC<CardBannerProps> = (props) => {
  return (
    <div
      data-testid="banner-id"
      className="banner-container"
      style={{
        backgroundColor: theme.palette.backgroundcolor.main,
        width: "100%",
        ...props.sx,
      }}
    >
      <Grid
        container
        // spacing={5}
        columnSpacing={{ xs: 3, sm: 6, md: 9 }}
        justifyContent="center"
        alignItems="center"
      >
        <Grid item container xs={7} spacing={10}>
          <Grid item>
            <TypographyText variant="heading" component="div">
              Explore Books on entrepreneurship
            </TypographyText>
          </Grid>
          <Grid item>
            <TypographyText
              color={theme.palette.textcolor.light}
              variant="subtitle2"
              component="div"
            >
              Everything you need to know about thriving on a shoestring budget,
              making your first million, and hiring right from the start.
            </TypographyText>
          </Grid>
        </Grid>
        <Grid item xs={4}>
          <Logo url="/res/logo/reading.png" />
        </Grid>
      </Grid>
    </div>
  );
};

export default CardBanner;
