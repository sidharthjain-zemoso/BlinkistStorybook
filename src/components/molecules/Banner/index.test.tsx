import { render, screen } from "@testing-library/react";
import Banner from "./";
import { ThemeProvider } from "@mui/material/styles";
import { theme } from "../../../themes";

test("renders a banner", () => {
  render(
    <ThemeProvider theme={theme}>
      <Banner />
    </ThemeProvider>
  );

  const bannerHeading = screen.getByText("Explore Books on entrepreneurship");
  const bannerText = screen.getByText(
    "Everything you need to know about thriving on a shoestring budget, making your first million, and hiring right from the start."
  );
  const bannerImage = screen.getByRole("img");
  expect(bannerHeading).toBeInTheDocument();
  expect(bannerText).toBeInTheDocument();
  expect(bannerImage).toBeInTheDocument();

  const banner = screen.getAllByTestId("banner-id");
  expect(banner).toMatchSnapshot();
});
