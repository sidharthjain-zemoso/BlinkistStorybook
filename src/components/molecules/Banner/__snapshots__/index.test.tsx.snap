// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a banner 1`] = `
Array [
  <div
    class="banner-container"
    data-testid="banner-id"
    style="background-color: rgb(241, 246, 244); width: 100%;"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-1cvp9eb-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-10 MuiGrid-grid-xs-7 css-1fs7rb-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiTypography-root MuiTypography-heading css-s7yxi6-MuiTypography-root"
          >
            Explore Books on entrepreneurship
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiTypography-root MuiTypography-subtitle2 css-13dyvr6-MuiTypography-root"
          >
            Everything you need to know about thriving on a shoestring budget, making your first million, and hiring right from the start.
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
      >
        <img
          alt="Logo"
          src="/res/logo/reading.png"
        />
      </div>
    </div>
  </div>,
]
`;
